avm/res/compute/virtual-machine/main.bicep

```diff
@@ -536,6 +536,10 @@ resource managedDataDisks 'Microsoft.Compute/disks@2024-03-02' = [
       diskMBpsReadWrite: dataDisk.?diskMBpsReadWrite
       publicNetworkAccess: publicNetworkAccess
       networkAccessPolicy: networkAccessPolicy
+      encryption: {
+        diskEncryptionSetId: dataDisk.managedDisk.?diskEncryptionSetResourceId
+        type: 'EncryptionAtRestWithCustomerKey'
```

Comment:
I guess this should rather be something akin to `dataDisk.managedDisk.?type ?? 'EncryptionAtRestWithCustomerKey`'?